orttraining/orttraining/core/graph/training_op_defs.cc

```diff
@@ -4571,6 +4571,36 @@ Return true if all elements are true and false otherwise.
           updateOutputShape(ctx, 6, {num_directions, three * hidden_size});
         }
       });
+
+  ONNX_CONTRIB_OPERATOR_SCHEMA(PadByAxis)
+      .SetDomain(kMSDomain)
+      .SinceVersion(1)
+      .SetDoc(
+          "PadByAxis operator pads zero on specific axis, and unflatten the dimension on axis to given unflatten_dims."
+          "This is used by padding elimination graph transformers.")
+      .Input(0, "input", "input data of rank N, shape is [d1, d2, ..., dN]", "T")
+      .Input(1, "indices", "1D Tensor of int32/int64 indices, indexing on axis.", "T_INDEX")
+      .Input(2, "unflatten_dims", "1D tensor with two values, [M1, M2].", "T_INT")
+      .Output(0, "output", "output data of rank N+1, [M1, M2, d2, ..., dN]", "T")
+      .Output(1, "unflatten_output_shape", "1D tensor with output shape, [M1*M2, d2, ..., dN]", "T_INT")
+      .Attr(
+          "axis",
```

Comment:
It seems the 'axis' it no use in implementation?

Reply:
I hesitated to add this attribute, maybe we can remove it until we need it. Then we would rename the op to PadAndUnflattenFirstAxis.

Reply:
maybe just add a 'TODO' ?